core/src/main/scala/io/projectglow/vcf/VCFFileWriter.scala

```diff
@@ -73,7 +73,20 @@ object VCFFileWriter extends GlowLogging {
     options.getOrElse(VCF_HEADER_KEY, defaultHeader.get) match {
       case INFER_HEADER =>
         logger.info("Inferring header for VCF writer")
-        (VCFSchemaInferrer.headerLinesFromSchema(schema).toSet, None)
+        val headerLines = VCFSchemaInferrer.headerLinesFromSchema(schema).toSet
+        if (options.contains("inferredSamples") || options.contains("numMissingSamples")) {
+          val inferredSamples =
+            options.get("inferredSamples").map(_.split("\t")).getOrElse(Array.empty)
+          val missingSamples = options
+            .get("numMissingSamples")
+            .map { n =>
+              Array.fill(n.toInt)("")
```

Comment:
Is this the same as the default sample ids that we inject in `VCFStreamWriter`?

Reply:
Also, it seems a little sketchy to me to inject sample ids if we don't have any indication that each row represents the same cohort. If each row has the same number of samples it makes sense to me, otherwise I think this is likely to return incomprehensible results.